modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -88,11 +88,42 @@ export function getValueDetailsFromItem(item, fieldName) {
 		return null;
 	}
 
+	// TODO: extract function to get same behavior as block 130-143
+	const languageId =
+		Liferay.ThemeDisplay.getLanguageId() ||
+		Liferay.ThemeDisplay.getBCP47LanguageId() ||
+		Liferay.ThemeDisplay.getDefaultLanguageId();
+
+	const rawTextFieldName = `${fieldName}RawText`;
+	const i18nFieldName = `${fieldName}_i18n`;
+
 	let rootPropertyName = fieldName;
 	const valuePath = [];
 	let navigatedValue = item;
 
-	if (Array.isArray(fieldName)) {
+	// Cover richText content value
+	// {name: '<p>...</p>', nameRawText: '...'}
+	if (
+		Object.hasOwn(item, rawTextFieldName) &&
```

Comment:
This is a bit opinionated imho, however we need to cover the case of a rich-text field in a generic way. We will need to revisit this to formally determine what do we want to do with cells containing html markup. Perhaps solution is a dedicated renderer. Ok for the moment

Reply:
Yeah, I agree, The ideal solution, in my opinion, is that the default renderer shows raw HTML, and new rich text renderer shows applied HTML. That can definitely be a separate task.